### DISCUSSION TITLE
In Laravel, all files from nova-components come with an ar.json file only, even if you are using the English (LTR) version.

### DISCUSSION THREAD

**Discussion by mohamedabuassi99** [laravel/nova-issues#6417] [Q&A] [answered]
- Laravel Version: 10.48.12
- Nova Version: 4.x
- PHP Version: 8.1
- Database Driver & Version: mysql
- Operating System and Version: os 14
- Browser type and version: chrome 125.0.6422.114

### Description:
my laravel app working with 2 differance language ar and en  with rtl and ltr so when i use rtl ar all works good  when i change to en ltr all my files inside nova-components working on ar.json (arabic) why this happend to me?

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by mohamedabuassi99** [comment]
How can I send you my private repository?

**Comment by crynobone** [comment]
RTL and language are two different features which doesn't depends on each other. You need to provide a reproducing repository in order for us to understand what's exactly is wrong and to debug the issue.

**Comment by mohamedabuassi99** [comment]
okay okay the main problem is my program take all the data from ar.json
while the app running en  why?

**Comment by mohamedabuassi99** [comment]
My problem is all the data inside nova-components are translated to ar.json
even if my app is en to get from en.json
I have to remove ar.json to get my english translation
why?

**Comment by mohamedabuassi99** [comment]
i put my problem inside this video
https://drive.google.com/file/d/1zT5TcbQ28b1JVhH0so5BP5YwTirt73Of/view?usp=sharing

**Comment by mohamedabuassi99** [comment]
look i go to app.php and change local to en it works with english and
ar.json not working
there is a problem with nova-components with this issue

**Comment by crynobone** [comment, marked as answer]
`HandleInertiaRequests` loads the translations for the frontend, you need to change the locale earlier than that.

```diff
    'middleware' => [
        'web',
+       \App\Http\Middleware\LanguageMiddleware::class,
        HandleInertiaRequests::class,
        DispatchServingNovaEvent::class,
        BootTools::class,
-       \App\Http\Middleware\LanguageMiddleware::class,
    ],
```
